src/System.Devices.Gpio/System.Devices.Gpio/UnixDriver.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Devices.Gpio
+{
+    public class UnixDriver : GpioDriver
+    {
+        private const string GpioPath = "/sys/class/gpio";
+
+        private ISet<int> _exportedPins;
+
+        public UnixDriver()
+        {
+            _exportedPins = new HashSet<int>();
+        }
+
+        public override void Dispose()
+        {
+            while (_exportedPins.Count > 0)
```

Comment:
At this stage this is a nit, more like something to look ahead: I am not sure that a HashSet and ISet are good abstractions here, perhaps, BitArray works better in this context - or depending on the number of pins accessing flags. Using a BitArray can let you use foreach instead of relying that UnexportPin remove it from the collection.

Reply:
The problem is that for this driver implementation (UnixDriver) that is meant to be generic enough to be used in all Unix based OS among different devices, I don't know if there is a way to get how many GPIO pins the device has, so I cannot use the BitArray if I don't know how many bits I will need.

Maybe the pin count should be a parameter the user will have to provide in the UnixDriver constructor.